### ISSUE TITLE
[BUG] Error on dragging multiple layers into a group

### ISSUE THREAD

**Issue by K-Meech** [brainglobe/napari-experimental#25]
**Describe the bug**
Open the plugin and add some layers (e.g. one points, one shapes, one labels). Click the button to add an empty layer group. Ctrl click both the points and labels and drag them onto the group. An error is thrown:

```
File ~\anaconda3\envs\napari-experimental\lib\site-packages\napari\utils\events\containers\_typed.py:149, in TypedMutableSequence.__getitem__(self=Node[SL9PS], key=3)
    146     except ValueError as e:
    147         raise KeyError(str(e)) from e
--> 149 result = self._list[key]
        self._list = [Node[Points], Node[Shapes], Node[X1JK7]]
        key = 3
        self = Node[SL9PS]
    150 return self.__newlike__(result) if isinstance(result, list) else result

IndexError: list index out of range
```

**Expected behaviour**
All the selected layers should be added to the group.

**Comment by willGraham01** [comment]
Interesting to note that when trying this, the _first_ item that should be moved is correctly moved, and the error seems to get raised when moving the next item:

![image](https://github.com/brainglobe/napari-experimental/assets/32364977/49577f75-9682-4d65-a141-afcb2f4fd40d)

Looking at the traceback, my theory is that this is because the destination indices do not change linearly as layers are moved one at a time.

Before move:
```
root
- Points
- Blobs
- Group[XXXXX]
```

Move multiple event is fired off: first task is to move `Points` INTO the `Group[XXXXX]` which is at position 2. This happens fine, and we then are left with

```
root
- Blobs
- Group[XXXXX]
  - Points
```

We now fire off the next move, which _has already been set_ to be moving `Blobs` into `Group[XXXXX]` at position 2. BUT `Group[XXXXX]` is no longer at position 2, because `Points` has been moved inside it! Thus, attempting to grab `self[2]` returns the key error (because we now want to grab `self[1]`, which is where `Group[XXXXX]` has inadvertently moved to). This behaviour is unique to the tree-like structures that we have, and would be fine if we just had a flat list.

**Comment by willGraham01** [comment]
Further reinforcing the above theory, following the steps in the issue, but moving the empty GroupLayer to the top of the GroupLayer list, and then selecting all the `Layers` and moving them into the Group does not raise an error. (Which makes sense, since the Group is staying in the same place).

Edit: further still, this error only seems to occur if at least one of the selected Layers is _above_ the GroupLayer they are to be moved into.

**Comment by willGraham01** [comment]
Closing as fixed in #28
